src/info.c

```diff
@@ -1273,6 +1273,43 @@ static void info_shiplogView( unsigned int wid, char *str )
       dialogue_msgRaw( _("Log message"), tmp);
 }
 
+/**
+ * @brief Asks the player for an entry to add to the log
+ *
+ * @param wid Window widget
+ * @param str Button widget name
+ */
+static void info_shiplogAdd( unsigned int wid, char *str )
+{
+   char *tmp;
+   char *logname;
+   int logid;
+   (void) str;
+
+   logname = toolkit_getList( wid, "lstLogs" );
+   if ( logname == NULL || !strcmp( "All", logname ) ){
+      tmp = dialogue_inputRaw( "Add a log entry", 0, 4096, "Add an entry to your diary:" );
+      if ( tmp != NULL && strlen ( tmp ) > 0 ){
```

Comment:
Need parentheses around each statement (what && separates) and there should not be a space after strlen.